python/publish/publisher.py

```diff
@@ -52,6 +52,13 @@ class Settings:
     check_run_annotation: List[str]
     seconds_between_github_reads: float
     seconds_between_github_writes: float
+    job_summary: bool
+
+    @property
+    def job_summary_file(self) -> Optional[str]:
+        filepath = os.getenv('GITHUB_STEP_SUMMARY')
+        if self.job_summary and filepath:
+            return filepath
```

Comment:
Lets handle the `job_summary_file` as all other settings:

```suggestion
    job_summary_file: Optional[str]
```

and validate its value in `get_settings` below.